dask/dataframe/io/parquet.py

```diff
@@ -375,13 +373,22 @@ def _read_pyarrow_parquet_piece(open_file_func, piece, columns, index_cols,
 
 def _write_pyarrow(df, path, write_index=None, append=False,
                    ignore_divisions=False, partition_on=None,
-                   storage_options=None, **kwargs):
+                   storage_options=None,
+                   row_group_size=None, version='1.0',
+                   use_dictionary=True, compression='snappy',
+                   use_deprecated_int96_timestamps=None,
+                   coerce_timestamps=None,
+                   flavor=None, chunk_size=None):
```

Comment:
The copying of the default values here is rather unfortunate, and fragile to changes to keyword default values. On second thought, I think I prefer my second solution I suggested previously. This diff seems to do the correct thing:

```diff
diff --git a/dask/dataframe/io/parquet.py b/dask/dataframe/io/parquet.py
index 36dc347..d4acf17 100644
--- a/dask/dataframe/io/parquet.py
+++ b/dask/dataframe/io/parquet.py
@@ -373,6 +373,14 @@ def _read_pyarrow_parquet_piece(open_file_func, piece, columns, index_cols,
         return df
 
 
+_write_table_kwargs = {'row_group_size', 'version', 'use_dictionary',
+                       'compression', 'use_deprecated_int96_timestamps',
+                       'coerce_timestamps', 'flavor', 'chunk_size'}
+
+_write_metadata_kwargs = {'version', 'use_deprecated_int96_timestamps',
+                          'coerce_timestamps'}
+
+
 def _write_pyarrow(df, path, write_index=None, append=False,
                    ignore_divisions=False, partition_on=None,
                    storage_options=None, **kwargs):
@@ -384,6 +392,11 @@ def _write_pyarrow(df, path, write_index=None, append=False,
         raise NotImplementedError("`partition_on` not implemented for "
                                   "`engine='pyarrow'`")
 
+    if set(kwargs).difference(_write_table_kwargs):
+        msg = ("Unexpected keyword arguments: "
+               "%r" % list(set(kwargs).difference(_write_table_kwargs)))
+        raise TypeError(msg)
+
     if write_index is None and df.known_divisions:
         write_index = True
 
@@ -413,8 +426,9 @@ def _write_partition_pyarrow(df, open_with, filename, write_index,
 
     if metadata_path is not None:
         with open_with(metadata_path, 'wb') as fil:
-            kwargs.pop('compression', None)
-            parquet.write_metadata(t.schema, fil, **kwargs)
+            kwargs2 = {k: v for k, v in kwargs.items()
+                       if k in _write_metadata_kwargs}
+            parquet.write_metadata(t.schema, fil, **kwargs2)
 
 
 # ----------------------------------------------------------------------
```

Reply:
A second benefit of doing the above is that we only encode user specified keywords in the tasks, rather than every value, making the tasks mildly smaller.